### PR TITLE
Add support for Aqara H1 Remote variants 3, 4 & 5

### PR DESCRIPTION
**Pull request by ShogunMan**

### CODE REVIEW

File: zhaquirks/tuya/ts0601_dimmer.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ class TuyaSingleSwitchDimmerGP(TuyaDimmerSwitch):
         MODELS_INFO: [
             ("_TZE200_3p5ydos3", "TS0601"),
             ("_TZE200_ip2akl4w", "TS0601"),
+            ("_TZE200_vucankjx", "TS0601"),  # Loratap
```
Comment (MattWestb) [inline zhaquirks/tuya/ts0601_dimmer.py:151]: Dont adding this one more time in this PR then you have doing it in the original and also the same for line 211.

**Review by MattWestb — COMMENTED** [review]
Flace8 is complaining : `Warning: zhaquirks/xiaomi/aqara/remote_h1.py:84:1: D101 Missing docstring in public class` but the dockstring looks OK for my and is is not strange formatted :-((

File: zhaquirks/tuya/ts0601_dimmer.py (hunk 1 above)
Comment (ShogunMan) [reply zhaquirks/tuya/ts0601_dimmer.py:151]: I accidentally pushed the wrong file, latest is the correct one.

**Comment by MattWestb** [comment]
From my point its looks OK only the the py test is failing then its not written and new test for the new devices that is needed for merging the code to the main brash and i hope some can helping writing that so we can getting  support for this devices.

Great that is working for your devices in your system and other user can using it as local quirk until is being fixed  :-))

### Great work done !!

File: zhaquirks/xiaomi/aqara/remote_h1.py [hunk 2]
```diff
@@ -1,294 +1,626 @@
...
+            4: {
+                # "4": {
+                #   "profile_id": 260, "device_type": "0x0103",
+                #   "in_clusters": ["0x0003, 0x0012"],
+                #   "out_clusters": ["0x0006"] },
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT_SWITCH,
+                INPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    MultistateInputCluster,
```
Comment (javicalle) [inline zhaquirks/xiaomi/aqara/remote_h1.py:462]: I believe that tests don't like this here. Maybe can be fixed using `MultistateInputCluster.cluster_id` instead, but are you sure that these clusters must be in the signature part?
Can you post the device signature (without the quirk)?

File: zhaquirks/xiaomi/aqara/remote_h1.py [hunk 3]
```diff
@@ -1,294 +1,626 @@
...
+            5: {
+                # "5": {
+                #   "profile_id": 260, "device_type": "0x0103",
+                #   "in_clusters": ["0x0003, 0x0012"],
+                #   "out_clusters": ["0x0006"] },
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT_SWITCH,
+                INPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    MultistateInputCluster,
```
Comment (javicalle) [inline zhaquirks/xiaomi/aqara/remote_h1.py:475]: Same that above.

File: zhaquirks/xiaomi/aqara/remote_h1.py [hunk 4]
```diff
@@ -1,294 +1,626 @@
...
+            6: {
+                # "6": {
+                #   "profile_id": 260, "device_type": "0x0103",
+                #   "in_clusters": ["0x0003, 0x0012"],
+                #   "out_clusters": ["0x0006"] },
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT_SWITCH,
+                INPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    MultistateInputCluster,
```
Comment (javicalle) [inline zhaquirks/xiaomi/aqara/remote_h1.py:488]: Same that above.

File: zhaquirks/xiaomi/aqara/remote_h1.py [hunk 5]
```diff
@@ -1,294 +1,626 @@
...
+            },
+        }
+    }
+    device_automation_triggers = RemoteH1DoubleRocker1.device_automation_triggers
+
+
+class RemoteH1DoubleRocker5(XiaomiCustomDevice):
+    """Aqara H1 Wireless Remote Double Rocker Version WRS-R02, variant 5."""
+
+    signature = {
```
Comment (javicalle) [inline zhaquirks/xiaomi/aqara/remote_h1.py:545]: That signature can't be fine.
Be sure to keep the device awake during all the pairing process.

Comment (ShogunMan) [reply zhaquirks/xiaomi/aqara/remote_h1.py:545]: This is the signature for my device:
```
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4447, maximum_buffer_size=127, maximum_incoming_transfer_size=100, server_mask=11264, maximum_outgoing_transfer_size=100, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0x0103",
      "in_clusters": [
        "0x0000",
        "0x0001",
        "0x0003",
        "0x0012",
        "0xfcc0"
      ],
      "out_clusters": [
        "0x0003",
        "0x0006"
      ]
    },
    "2": {
      "profile_id": 260,
      "device_type": "0x0103",
      "in_clusters": [
        "0x0003",
        "0x0012"
      ],
      "out_clusters": [
        "0x0003",
        "0x0006"
      ]
    },
    "3": {
      "profile_id": 260,
      "device_type": "0x0103",
      "in_clusters": [
        "0x0003",
        "0x0012"
      ],
      "out_clusters": [
        "0x0003",
        "0x0006"
      ]
    }
  },
  "manufacturer": "LUMI",
  "model": "lumi.remote.b28ac1",
  "class": "remote_h1.RemoteH1DoubleRocker2"
}
```

Comment (ShogunMan) [reply zhaquirks/xiaomi/aqara/remote_h1.py:545]: All that I know is that this quirk is working with my device.

Comment (javicalle) [reply zhaquirks/xiaomi/aqara/remote_h1.py:545]: That signature is from a quirked device.
I'm not going to say that this quirk isn't working with your device (which uses the `RemoteH1DoubleRocker2`), but this signature can't be correct on any device:
```python
class RemoteH1DoubleRocker5(XiaomiCustomDevice):
    """Aqara H1 Wireless Remote Double Rocker Version WRS-R02, variant 5."""

    signature = {
        MODELS_INFO: [(None, None)],
        ENDPOINTS: {
            1: {
                # "1": {
                #   "profile_id": 260, "device_type": "0x0105",
                #   "in_clusters": [],
                #   "out_clusters": [] }
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.COLOR_DIMMER_SWITCH,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [],
            },
            2: {
                # "2": {
                #   "profile_id": 260, "device_type": "0x0103",
                #   "in_clusters": [],
                #   "out_clusters": [] },
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT_SWITCH,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [],
            },
            3: {
                # "3": {
                #   "profile_id": 260, "device_type": "0x0103",
                #   "in_clusters": [],
                #   "out_clusters": [] },
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT_SWITCH,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [],
            },
        },
    }
```

It seems to me to be an incomplete paired device.

**Comment by javicalle** [comment]
> @javicalle We have getting the normal test working OK but some "run tests" i cant fixing is it the real code test that need being implanted ?

I believe that is related to the use of `MultistateInputCluster` (without the `.cluster_id`) in the `signature` part. It would be fine for the `replacement` part, but I think that tests complains if used in the `signature`.

**Comment by ShogunMan** [comment]
Some of this is covered in the following open issue:
[](https://github.com/zigpy/zha-device-handlers/issues/940)

Maybe this can be of some help.

File: zhaquirks/xiaomi/aqara/remote_h1.py (hunk 5 above)
Comment (ShogunMan) [reply zhaquirks/xiaomi/aqara/remote_h1.py:545]: I'll see if I can provide the signature for the 'unquirked' device.

**Pull request closed without merging**